File: backend-JS/src/controllers/roomController.js
const { roomService } = require('../services');
const { success, created, error } = require('../utils/response');

const getPublicRooms = async (req, res) => {
  try {
    const rooms = await roomService.getPublicRooms();
    return success(res, { rooms, total: rooms.length }, '获取成功');
  } catch (err) {
    return error(res, 500, '获取房间失败');
  }
};

const getMyRooms = async (req, res) => {
  try {
    const rooms = await roomService.getMyRooms(req.userId);
    return success(res, { rooms, total: rooms.length }, '获取成功');
  } catch (err) {
    return error(res, 500, '获取房间失败');
  }
};

const getRoom = async (req, res) => {
  try {
    const room = await roomService.getRoom(parseInt(req.params.id), req.userId);
    return success(res, room, '获取成功');
  } catch (err) {
    return error(res, 404, err.message);
  }
};

const createRoom = async (req, res) => {
  try {
    const room = await roomService.createRoom(req.userId, req.body);
    return created(res, room, '创建成功');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const deleteRoom = async (req, res) => {
  try {
    await roomService.deleteRoom(parseInt(req.params.id), req.userId);
    return success(res, null, '房间已删除');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const joinRoom = async (req, res) => {
  try {
    await roomService.joinRoom(parseInt(req.params.id), req.userId);
    return success(res, null, '加入成功');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const leaveRoom = async (req, res) => {
  try {
    await roomService.leaveRoom(parseInt(req.params.id), req.userId);
    return success(res, null, '已离开房间');
  } catch (err) {
    return error(res, 400, err.message);
  }
};

const getRoomMembers = async (req, res) => {
  try {
    const members = await roomService.getRoomMembers(parseInt(req.params.id));
    return success(res, { members }, '获取成功');
  } catch (err) {
    return error(res, 500, '获取成员失败');
  }
};

const getRoomMessages = async (req, res) => {
  try {
    const page = parseInt(req.query.page) || 1;
    const limit = parseInt(req.query.limit) || 50;
    
    const result = await roomService.getRoomMessages(parseInt(req.params.id), page, limit);
    return success(res, result, '获取成功');
  } catch (err) {
    return error(res, 500, '获取消息失败');
  }
};

const sendRoomMessage = async (req, res) => {
  try {
    const { content } = req.body;
    
    if (!content) {
      return error(res, 400, '内容不能为空');
    }
    
    const message = await roomService.sendRoomMessage(parseInt(req.params.id), req.userId, content);
    return success(res, message, '发送成功');
  } catch (err) {
    return error(res, 500, '发送消息失败');
  }
};

module.exports = {
  getPublicRooms,
  getMyRooms,
  getRoom,
  createRoom,
  deleteRoom,
  joinRoom,
  leaveRoom,
  getRoomMembers,
  getRoomMessages,
  sendRoomMessage
};

File: backend-C#/Controllers/RoomController.cs
using IMS.DTOs;
using IMS.DTOs.Requests;
using IMS.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace IMS.Controllers;

[ApiController]
[Route("api/social/rooms")]
[Authorize]
public class RoomController : BaseController
{
    private readonly IRoomService _roomService;

    public RoomController(IRoomService roomService)
    {
        _roomService = roomService;
    }

    [HttpGet("public")]
    public async Task<ActionResult<ApiResponse>> GetPublicRooms()
    {
        var result = await _roomService.GetPublicRoomsAsync();
        return Success(new { rooms = result.Rooms, total = result.Total }, "获取成功");
    }

    [HttpGet("my")]
    public async Task<ActionResult<ApiResponse>> GetMyRooms()
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        var result = await _roomService.GetMyRoomsAsync(CurrentUserId.Value);
        return Success(new { rooms = result.Rooms, total = result.Total }, "获取成功");
    }

    [HttpGet("{id}")]
    public async Task<ActionResult<ApiResponse>> GetRoom(long id)
    {
        var room = await _roomService.GetRoomAsync(id);
        if (room == null)
        {
            return Error("房间不存在", 404);
        }

        return Success(room, "获取成功");
    }

    [HttpPost]
    public async Task<ActionResult<ApiResponse>> CreateRoom([FromBody] CreateRoomRequest request)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            var room = await _roomService.CreateRoomAsync(CurrentUserId.Value, request);
            return Success(room, "创建成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPut("{id}")]
    public async Task<ActionResult<ApiResponse>> UpdateRoom(long id)
    {
        // Placeholder
        return Success(new { }, "更新成功");
    }

    [HttpDelete("{id}")]
    public async Task<ActionResult<ApiResponse>> DeleteRoom(long id)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            await _roomService.DeleteRoomAsync(CurrentUserId.Value, CurrentUserRole ?? "", id);
            return Success("删除成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPost("{id}/join")]
    public async Task<ActionResult<ApiResponse>> JoinRoom(long id)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            await _roomService.JoinRoomAsync(CurrentUserId.Value, id);
            return Success("加入成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPost("{id}/leave")]
    public async Task<ActionResult<ApiResponse>> LeaveRoom(long id)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            await _roomService.LeaveRoomAsync(CurrentUserId.Value, id);
            return Success("离开成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpGet("{id}/members")]
    public async Task<ActionResult<ApiResponse>> GetRoomMembers(long id)
    {
        try
        {
            var result = await _roomService.GetRoomMembersAsync(id);
            return Success(new { members = result.Members, total = result.Total }, "获取成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpGet("{id}/messages")]
    public async Task<ActionResult<ApiResponse>> GetRoomMessages(long id, [FromQuery] RoomMessagesRequest request)
    {
        try
        {
            var result = await _roomService.GetRoomMessagesAsync(id, request);
            return Success(new 
            { 
                messages = result.Messages, 
                total = result.Total, 
                page = result.Page, 
                limit = result.Limit 
            }, "获取成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }

    [HttpPost("{id}/messages")]
    public async Task<ActionResult<ApiResponse>> SendRoomMessage(long id, [FromBody] SendRoomMessageRequest request)
    {
        if (!CurrentUserId.HasValue)
        {
            return Error("未登录", 401);
        }

        try
        {
            var message = await _roomService.SendRoomMessageAsync(CurrentUserId.Value, id, request);
            return Success(message, "发送成功");
        }
        catch (InvalidOperationException ex)
        {
            return Error(ex.Message, 400);
        }
    }
}
